rfc/php_technical_committee.wiki

```diff
@@ -0,0 +1,183 @@
+  - The TC Secretary MUST announce the decision and reasoning for it on the issue in question and remove the `tc-hold` tag.
+  - The decision of the TC is binding.  That means a relevant change MUST NOT be merged until and unless it conforms with the TC's decision.  Depending on the decision, the core developer involved may merge the issue, close it, or revise it accordingly.
+
+If an issue has been marked `tc-hold` for more than two weeks with no response or acknowledgement from the TC, the
+developer SHOULD post to the Internals mailing list a notification of absence.
+
+If a notification of absence receives no response from the TC or any member of it for one month, then the entire TC
+is considered AWOL and removed.  New elections must then be immediately called as described above.
+
+The election of a new TC will "reset the clock" on any outstanding `tc-hold` issues.
+
+==== Decision process ====
+
+=== User-facing changes ===
+
+The RFC process will continue to be used for user facing changes.  However, it will no longer be used for
+non-user-facing technical decisions.
+
+TC members have no special role or authority in the RFC process.
+
+=== Process changes ===
+
+The RFC process will continue to be used for process changes, including changes to the role of the TC.
+
+=== Technical changes ===
+
+As noted, the TC has no special role or authority on user-facing changes.  The TC may, however, decide on the
+implementation of those features. In other words the TC can decide about implementation and C ABI and API but
+not about PHP API.
+
+The TC SHOULD NOT block the merging of a user-facing change approved by the RFC process, unless the provided
+implementation would result in introduction of new bugs, side effects not mentioned in the RFC, significant performance
+penalties not mentioned in RFC, or if there is an equivalent implementation in progress that TC finds more appropriate.
+
+The TC SHOULD give due consideration and weight to previous decisions regarding technical aspects, either by the TC
+or by RFCs prior to the TC's creation.  However, it is NOT REQUIRED to abide by them, and may make decisions as it
+sees best to ensure the quality and maintainability of the code at the time the decision is made.
+
+Technical aspects on which the TC may make a decision include, but are not limited to:
+
+  * Whether a given pull request implements a feature or resolves a bug.
+  * Whether a given change qualifies as user-facing, and thus requires an RFC.
+  * Whether a given technical aspect or implementation approach is acceptable.
+
+=== TC voting ===
+
+Once an issue has been brought to the TC, the Secretary must allow all TC members time to offer their input within one
+week.  Once the Secretary feels the issue has been sufficiently discussed, the Secretary will call a vote of the TC
+members. The vote will be private and conducted however the TC chooses.
```

Comment:
The one-week time frame might be a bit short, and also not take into account TC members going on holiday.

Should there be a process for announcing temporary leave? Possibly via an email announcement on internals or a new dedicated TC-only list?

Reply:
What I want to be careful of here, and the wording likely needs to be tweaked, is "it's been 48 hours and all 5 of us agree, so we'll just announce a decision now and be done with it."  Which I expect to be the most common case.

The timeframe here is intended for "if Pete hasn't commented in X time, we'll just go ahead without him, he's had his chance."  I'm OK with one or two weeks here.

Maybe rephrase as "discussion must last at least 2 weeks or until all TC members have been able to comment, whichever comes first"?

Reply:
Basically, I want to avoid every time the TC does something it taking a full month.  I expect *most* issues will be "all 5 of us agree, cool? cool, I'll post it" within a day or three, and I don't want to have too much process to block that.

Reply:
> Basically, I want to avoid every time the TC does something it taking a full month. I expect _most_ issues will be "all 5 of us agree, cool? cool, I'll post it" within a day or three, and I don't want to have too much process to block that.

Yeah, I expect this to be the usual time frame and releasing answers quickly is better than always waiting 2 weeks. It's more I think the discussion taking *at most* 1 week is a bit short which is why I think 2 is better.

Reply:
So this might be getting slightly confusing. We currently have this for discussion:

> Once an issue has been brought to the TC, the Secretary must allow all TC members time to offer their input within one week.  Once the Secretary feels the issue has been sufficiently discussed, the Secretary will call a vote of the TC
members

I think this covers the shorter period as there is _within one week_. Would you prefer to rephrase this part and extend it to 2 weeks?

Then we have 2 weeks for voting

> The vote will last for two weeks or until all TC members have voted, whichever comes first.

So it seems to me that this covers shorter period as well unless someone is on holiday or busy and does not explicitly abstain. We could maybe put something that it is possible to vote before voting starts by letting secretary know in case someone goes on holiday before the vote. What do you think?